galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,288 @@
+					}
+				}
+				
+			}
+			
+			runIdsInWorklist.removeAll(runIdsNotFound);
+		}
+		
+		return runs;
+	}
+	
+	protected IRunResult getRunById(String id) throws ResultArchiveStoreException {
+
+		IRunResult run = null;
+	       
+        for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
+           
+           run = directoryService.getRunById(id);
+           
+           if(run != null) {
+        	   return run;
+           }
+           
+        }
+		
+		return null;
+	}
+	
+	protected void removeRunIdFromDss(String principalUsername, String runId) throws DynamicStatusStoreException, ServletException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user." + principalUsername + ".worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			JsonArray runIdsToRemove = new JsonArray();
+			for (JsonElement el : runIds) {
+				if (el.getAsString().equals(runId)) {
+					runIdsToRemove.add(el);
+				}
+			}
+			
+			if (runIdsToRemove.size()>0) {
+				for (JsonElement el : runIdsToRemove) {
+					runIds.remove(el);
+				}
+			}
+		
+			jsonObject.remove("runIds");
+			jsonObject.add("runIds", runIds);
+
+			
+			String newValue = jsonObject.toString();
+			
+			boolean exceptionThrown = false;
+			int count = 0;
+			
+			do {
+
+				try {
+					dss.performActions(
+							new DssSwap("user." + principalUsername + ".worklist", oldValue, newValue));
+				} catch (DynamicStatusStoreException e) {
+					exceptionThrown = true;
+					oldValue = dss.get("user." + principalUsername + ".worklist");
+					count++;
+					
+					if (count > 1000) {
+						throw new ServletException("Error while updating the DSS, maximum attempts reached");
+					}
+				}
+				
+				exceptionThrown = false;
+				
+			} while (exceptionThrown);
+			
+		}
+			
+	}
+
+	
+	protected void addRunIdToDss(String principalUsername, String runId) throws DynamicStatusStoreException, ServletException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user." + principalUsername + ".worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			boolean runIdInArray = false;
+			for (JsonElement el : runIds) {
+				if (el.getAsString().equals(runId)) {
+					runIdInArray = true;
+					break;
+				}
+			}
+			
+			if (runIdInArray == false) {
+				runIds.add(runId);
+			}
+			
+			jsonObject.remove("runIds");
+			jsonObject.add("runIds", runIds);	
+			
+			
+			String newValue = jsonObject.toString();
+			
+			boolean exceptionThrown = false;
+			int count = 0;
+			
+			do {
```

Comment:
this loop is wrong,  as with the method above.  it should include the above code as you want to reapply the change to the new "oldvalue"